nncf/tensorflow/sparsity/magnitude/algorithm.py

```diff
@@ -218,14 +218,13 @@ def _set_masks_for_threshold(self, threshold_val):
                         )
 
     def _collect_all_weights(self):
-        all_weights = []
-        for wrapped_layer in collect_wrapped_layers(self._model):
-            for weight_attr, ops in wrapped_layer.weights_attr_ops.items():
-                for op_name in ops:
-                    if op_name in self._op_names:
-                        all_weights.append(
-                            tf.reshape(self._weight_importance_fn(wrapped_layer.layer_weights[weight_attr]), [-1])
-                        )
+        all_weights = [
+            tf.reshape(self._weight_importance_fn(wrapped_layer.layer_weights[weight_attr]), [-1])
+            for wrapped_layer in collect_wrapped_layers(self._model)
```

Comment:
I'm not sure that we should push developers write such constructions. If you can customize the rule, I would suggest limit number of nested loops =1 or postpone these changes.